src/Environment.php

```diff
@@ -0,0 +1,99 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\Markdown;
+
+final class Environment<T> {
+  public function __construct(
+    private ParserContext $parser,
+    private RenderContext $context,
+    private Renderer<T> $renderer,
+  ) {}
+
+  public static function html(bool $unsafe = false): Environment<string> {
```

Comment:
In general, we consider bool parameters to be an antipattern

For this particular one, we definitely want the obvious explicit naming at the ca llers

Reply:
This would probably also be better as a function than a static method - then new ones can be added by third-party libraries and created in a separate library

Reply:
Agreed, I think this coud be split into 2 functions, `html` and `unsafeHtml` or something like that, instead of the bool flag.